build_projects/dotnet-cli-build/BuildFPMToolPreReqs.cs

```diff
@@ -0,0 +1,319 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// This task prepares the command line parameters for running a RPM build using FPM tool and also updates the copyright and changelog file tokens.
+    /// If parses various values from the config json by first reading it into a model and then builds the required string for parameters and passes it back.
+    /// 
+    /// </summary>
+    public class BuildFPMToolPreReqs : Task
+    {
+        [Required]
+        public string InputDir { get; set; }
+        [Required]
+        public string OutputDir { get; set; }
+        [Required]
+        public string PackageVersion { get; set; }
+        [Required]
+        public string ConfigJsonFile { get; set; }
+        [Output]
+        public string FPMParameters { get; set; }
+        public override bool Execute()
+        {
+            try
+            {
+                if (!File.Exists(ConfigJsonFile))
+                {
+                    throw new FileNotFoundException($"Expected file {ConfigJsonFile} was not found.");
+                }
+
+                // Open the Config Json and read the values into the model
+                TextReader projectFileReader = File.OpenText(ConfigJsonFile);
+                if (projectFileReader != null)
+                {
+                    string jsonFileText = projectFileReader.ReadToEnd();
+                    ConfigJson configJson = JsonConvert.DeserializeObject<ConfigJson>(jsonFileText);
+
+                    // Update the Changelog and Copyright files by replacing tokens with values from config json
+                    UpdateChangelog(configJson, PackageVersion);
+                    UpdateCopyRight(configJson);
+
+                    // Build the full list of parameters 
+                    FPMParameters = BuildCmdParameters(configJson, PackageVersion);
+                    Log.LogMessage(MessageImportance.Normal, "Generated RPM paramters:  " + FPMParameters);
+                }
+                else
+                {
+                    throw new IOException($"Could not open the file {ConfigJsonFile} for reading.");
+                }
+            }
+            catch (Exception e)
+            {
+                Log.LogError("Exception while processing RPM paramters: " + e.Message);
+            }
```

Comment:
It's not helpful to log only e.Message for arbitrary exceptions. If there's a hard to reproduce issue/bug, we'll want the inner exceptions and full stack trace.  I'd just let the let exceptions go and msbuild will log everything and fail the task.